### PR TITLE
Fix 0 to nbsp conversion in the listing module

### LINKED ISSUE #4856: &nbsp; statt 0 in Modul "Auflistung"
### Affected version(s)

4.9.30

### Description

Wenn man über das Auflistungs-Modul Werte einer Spalte versucht auszugeben, wird statt `0` irgendwie `&nbsp;` ausgegeben.

[https://github.com/contao/contao/blob/47697073752ae4d6fa6b63b28e0b54c50a8143f3/listing-bundle/src/Resources/contao/modules/ModuleListing.php#L316](https://github.com/contao/contao/blob/47697073752ae4d6fa6b63b28e0b54c50a8143f3/listing-bundle/src/Resources/contao/modules/ModuleListing.php#L316)

Aktuell nur in 4.9.30 gesehn und getestet.

### PR DESCRIPTION
**Pull request by bezin**
Fixes #4856

### CODE REVIEW

**Comment by bezin** [comment]
Needs to be merged into 4.13 and 5.x as well.

**Comment by fritzmg** [comment]
> Needs to be merged into 4.13 and 5.x as well.

Any fixes in 4.9 will always be merged upstream (unless otherwise not applicable).

**Review by ausi — DISMISSED** [review]

File: listing-bundle/src/Resources/contao/modules/ModuleListing.php [hunk 1]
```diff
@@ -417,6 +417,11 @@ protected function formatValue($k, $value, $blnListSingle=false)
 	{
 		$value = StringUtil::deserialize($value);
 
+		// Handle falsy values (see #4858)
+		if ($value === '0' || $value === 0 || $value === false) {
```
Comment (aschempp) [inline listing-bundle/src/Resources/contao/modules/ModuleListing.php]: Pretty sure this will raise an array casting warning if `$value` is an array

Comment (ausi) [reply listing-bundle/src/Resources/contao/modules/ModuleListing.php]: There is no string casting in this line? Are you referring to line 322?

Comment (ausi) [reply listing-bundle/src/Resources/contao/modules/ModuleListing.php]: And if you mean 322, the `$value` there is always a scalar. Can you point to where exactly the casting warning occurs?

**Review by ausi — DISMISSED** [review]

File: listing-bundle/src/Resources/contao/modules/ModuleListing.php (hunk 1 above)
Comment (fritzmg) [reply listing-bundle/src/Resources/contao/modules/ModuleListing.php]: Even if `$value` would have been an array for whatever reason: the cast to string will happen anyway in the template.

**Review by ausi — APPROVED** [review]

File: listing-bundle/src/Resources/contao/modules/ModuleListing.php (hunk 1 above)
Comment (leofeyer) [reply listing-bundle/src/Resources/contao/modules/ModuleListing.php]: Closing this as false assumption. https://3v4l.org/iYnjO

**Comment by leofeyer** [comment]
Thank you @bezin.

**Pull request merged**